src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java

```diff
@@ -180,6 +193,47 @@ public void testCreateAnomalyDetectorWithDuplicateName() throws Exception {
             );
     }
 
+    public void testCreateAnomalyDetectorWithFlattenedResultIndex() throws Exception {
+        AnomalyDetector detector = createIndexAndGetAnomalyDetector(
+            INDEX_NAME,
+            ImmutableList.of(TestHelpers.randomFeature(true)),
+            false,
+            true
+        );
+
+        // test behavior when AD is disabled
+        updateClusterSettings(ADEnabledSetting.AD_ENABLED, false);
+        Exception ex = expectThrows(
+            ResponseException.class,
+            () -> TestHelpers
+                .makeRequest(
+                    client(),
+                    "POST",
+                    TestHelpers.AD_BASE_DETECTORS_URI,
+                    ImmutableMap.of(),
+                    TestHelpers.toHttpEntity(detector),
+                    null
+                )
+        );
+        assertThat(ex.getMessage(), containsString(ADCommonMessages.DISABLED_ERR_MSG));
+
+        // test behavior when AD is enabled
+        updateClusterSettings(ADEnabledSetting.AD_ENABLED, true);
+        Response response = TestHelpers
+            .makeRequest(client(), "POST", TestHelpers.AD_BASE_DETECTORS_URI, ImmutableMap.of(), TestHelpers.toHttpEntity(detector), null);
+        assertEquals("Create anomaly detector with flattened result index failed", RestStatus.CREATED, TestHelpers.restStatus(response));
+        Map<String, Object> responseMap = entityAsMap(response);
+        String id = (String) responseMap.get("_id");
+        int version = (int) responseMap.get("_version");
+        assertNotEquals("response is missing Id", AnomalyDetector.NO_ID, id);
+        assertTrue("incorrect version", version > 0);
+        // ensure the flattened result index was created
+        String expectedFlattenedIndex = String
+            .format(Locale.ROOT, "opensearch-ad-plugin-result-test_flattened_%s", id.toLowerCase(Locale.ROOT));
+        boolean indexExists = indexExists(expectedFlattenedIndex);
+        assertTrue(indexExists);
```

Comment:
Can you also check when writing to custom result index, the flattened index gets correct flattened value?

Reply:
I can confirm the flattened index gets correct flattened value. I can include some screenshot from osd visualization page to show this

Reply:
I meant to include that in IT. It is a guardrail against future changes.

Reply:
got cha, will add another assertion check on the actual flattened fields

Reply:
updated the test case to include more assertion checks on the flattened result index mapping to ensure flattened fields are present